src/cdp/connection.ts

```diff
@@ -309,7 +309,8 @@ class CDPSession {
   }
 
   _processQueue() {
-    setTimeout(() => {
+    const waitWrapper = makeWaitForNextTask();
```

Comment:
I think it is important to call makeWaitForNextTask() once, and use the same wrapper all the time. Otherwise, `spinning` flag will be a new variable all the time.